pkg/meta/model/table.go

```diff
@@ -854,6 +858,96 @@ func (pi *PartitionInfo) GetPartitionIDByName(partitionDefinitionName string) in
 	return -1
 }
 
+// GetDefaultListPartition return the index of Definitions
+// that contains the LIST Default partition otherwise it returns -1
+func (pi *PartitionInfo) GetDefaultListPartition() int {
+	if pi.Type != model.PartitionTypeList {
+		return -1
+	}
+	defs := pi.Definitions
+	for i := range defs {
+		if len(defs[i].InValues) == 0 {
+			return i
+		}
+		for _, vs := range defs[i].InValues {
+			if len(vs) == 1 && vs[0] == "DEFAULT" {
+				return i
+			}
+		}
+	}
+
+	return -1
+}
+
+// CanHaveOverlappingDroppingPartition returns true if special handling is needed
+// during DDL of partitioned tables, where range or list with default partition
+// can have overlapping partitions.
+func (pi *PartitionInfo) CanHaveOverlappingDroppingPartition() bool {
+	if pi.DDLAction == ActionDropTablePartition &&
+		pi.DDLState == StateWriteOnly {
+		return true
+	}
+	return false
+}
+
+// ReplaceWithOverlappingPartitionIdx returns the overlapping partition
+// if there is one and a previous error.
+// Functions based on locatePartitionCommon, like GetPartitionIdxByRow
+// will return the found partition, with an error, since it is being dropped.
+// This function will correct the partition index and error if it can.
+func (pi *PartitionInfo) ReplaceWithOverlappingPartitionIdx(idx int, err error) (int, error) {
```

Comment:
Could you add a concrete example here to help understand what this function is doing?

Reply:
Especially more comments to explain the term `OverlappingDropping` here.

Reply:
Hi, I have updated the function comments, with an example of OverlappingDropping connected to CanHaveOverlappingDroppingPartition. Please read through it if it makes sense?